templates/tftmpl/condition.go

```diff
@@ -202,3 +202,83 @@ variable "catalog_services" {
   type = map(list(string))
 }
 `)
+
+type NodesCondition struct {
+	Datacenter string
+	Filter     string
+}
+
+func (n NodesCondition) SourceIncludesVariable() bool {
+	return true
+}
+
+func (n NodesCondition) ServicesAppended() bool {
+	return false
+}
+
+func (n NodesCondition) appendModuleAttribute(body *hclwrite.Body) {
+	body.SetAttributeTraversal("nodes", hcl.Traversal{
+		hcl.TraverseRoot{Name: "var"},
+		hcl.TraverseAttr{Name: "nodes"},
+	})
+}
+
+func (n NodesCondition) appendTemplate(w io.Writer) error {
+	q := n.hcatQuery()
+	_, err := fmt.Fprintf(w, nodesRegexTmpl, q)
+	if err != nil {
+		log.Printf("[ERR] (templates.tftmpl) unable to write nodes condition template")
+		return err
+	}
+
+	return nil
+}
+
+func (n NodesCondition) appendVariable(w io.Writer) error {
+	_, err := w.Write(variableNodes)
+	return err
+}
+
+func (n NodesCondition) hcatQuery() string {
+	var opts []string
+
+	if n.Datacenter != "" {
+		opts = append(opts, fmt.Sprintf("datacenter=%s", n.Datacenter))
+	}
+
+	if n.Filter != "" {
+		opts = append(opts, fmt.Sprintf("filter=%s", n.Filter))
+	}
+
+	if len(opts) > 0 {
+		return `"` + strings.Join(opts, `" "`) + `" ` // deliberate space at end
+	}
+	return ""
+}
+
+const nodesRegexTmpl = `
+nodes = [
```

Comment:
The attributes in the variable object look good 👍

We have a larger design choice for CTS to generate input variables as maps instead of lists for the purpose of modules to easily consume it with the [Terraform `for_each` syntax](https://www.terraform.io/docs/language/meta-arguments/for_each.html). Would you mind changing it from a list to map?

As for the key of the map, we could use `id.node.datacenter` for uniqueness. Similar to the ID of the [`var.services` input variable](https://github.com/hashicorp/consul-terraform-sync/blob/afe4d8d1225378fa81fcb5f7dad37ab71fa5295d/templates/tftmpl/tfvars.go#L98)